src/horizon_client.rs

```diff
@@ -1948,4 +2051,39 @@ mod tests {
             "account_debited"
         );
     }
+
+    #[tokio::test]
+    async fn test_get_all_liquidity_pools() {
+        let horizon_client =
+            HorizonClient::new("https://horizon-testnet.stellar.org".to_string()).unwrap();
+
+        let all_liquidity_pools_request = AllLiquidityPoolsRequest::new()
+            .add_native_reserve()
+            .add_alphanumeric4_reserve(
+                "USDC".to_string(),
+                "GBBD47IF6LWK7P7MDEVSCWR7DPUWV3NY3DTQEVFL4NAT4AQH3ZLLFLA5".to_string(),
+            )
+            .set_limit(2);
+
+        let all_liquidity_pools_response = horizon_client
+            .get_all_liquidity_pools(&all_liquidity_pools_request)
+            .await;
+
+        assert!(all_liquidity_pools_response.clone().is_ok());
+    }
+
+    #[tokio::test]
+    async fn test_get_single_liquidity_pool() {
+        let horizon_client =
+            HorizonClient::new("https://horizon-testnet.stellar.org".to_string()).unwrap();
+
+        let single_liquidity_pool_request = SingleLiquidityPoolRequest::new()
+            .set_liquidity_pool_id("01c58ab8fb283c8b083a26bf2fe06b7b6c6304c13f9d29d956cdf15a48bea72d".to_string()).unwrap();
+
+        let single_liquidity_pool_response = horizon_client
+            .get_single_liquidity_pool(&single_liquidity_pool_request)
+            .await;
+
+        assert!(single_liquidity_pool_response.clone().is_ok());
```

Comment:
Tevens zou ik wat checks toevoegen op de inhoude van de response. Zitten de velden erin dit ik verwacht? Deserialized alles goed? Een `is_ok()` assert is wel erg minimaal